mempool/clist_mempool.go

```diff
@@ -47,11 +48,21 @@ type CListMempool struct {
 	// Keeps track of the rechecking process.
 	recheck *recheck
 
-	// Data in `txs` and `txsMap` must to be kept in sync and updated atomically.
+	// Data in the following variables must to be kept in sync and updated atomically.
 	txsMtx   cmtsync.RWMutex
-	txs      *clist.CList                    // concurrent linked-list of valid txs
-	txsMap   map[types.TxKey]*clist.CElement // for quick access to txs
+	lanes    map[types.Lane]*clist.CList     // each lane is a linked-list of (valid) txs
```

Comment:
Do we still need `CList` (i.e., "concurrent" list)?
If the answer if no, we probably need to change it in a further PR (shouldn't block this one).

Reply:
yeah, I would like to replace `clist.CList` by `list.List` in a future PR. There's still one thing that we need from CList, which is that `Len()` is guarded by an internal mutex in CList but not in List, so that creates race conditions.